evil-tests.el

```diff
@@ -7260,7 +7265,14 @@ Tiny ln"
      "Short 
 A much
 A me[d]
-Tiny ")))
+Tiny "))
+  (ert-info ("Visual block does not extend up to curswant unless tracking EOL")
+    (evil-test-buffer
+     "[f]oo bar
+foo"
+     ("\C-vfrjd")
+     "[ ]bar
```

Comment:
I tried this in vim and see that the space before `bar` is also deleted. With this, just before I delete I see that the cursor is "beyond-eol" on the second line (which is fine), but the block selection stops at the end of foo, rather than on the space.

Reply:
I think that is a separate issue where `evil-visual-range` results are the same beyond-eol and not, since you can replicate it even without this patch if you follow "j" with "hl" to make it ignore `temporary-goal-column`. 

Would need to change [this code](https://github.com/emacs-evil/evil/blob/5ce46a1fc175a8f13507ce2b6ec4c3618923f093/evil-types.el#L175-L208) to allow returning an inclusive range. Perhaps that could be delegated to a follow-up PR, since I do not know how you'd go about doing that, and since this PR at least does not make that matter worse.

Reply:
Sure, a separate PR is fine. But we shouldn't add tests which assert incorrect behaviour. So either we remove the test or we fix that first, then merge this. It's a test I think we should have, so I'm inclined to favour the latter. I may have time to take a look this weekend if you don't beat me to it.

Reply:
I changed the test now so that the incorrect behavior is not exhibited, because I would be positively surprised if that ever gets fixed.